server/routes/return-of-service.js

```diff
@@ -98,29 +118,88 @@ router.get(
     keycloak.allowRolesMiddleware('ministry_of_health', 'health_authority', 'employer')
   ),
   asyncMiddleware(async (req, res) => {
+    const actionName = 'ros-status-get';
     const { participantId } = req.params;
     const { email, user_id: userId, sub: localUserId } = req.user;
     const user = userId || localUserId;
     if (!(email && user)) {
       return res.status(401).send('Unauthorized user');
     }
-    // Validate Participant
-    const [participant] = await getParticipantByID({ id: participantId });
+    const participant = await validateParticipant(participantId, actionName);
     if (!participant) {
-      logger.error({
-        action: 'ros-status-create',
-        message: `Participant ${participantId} not found`,
-      });
       return res.status(404).send('Participant not found');
     }
     const resp = await getReturnOfServiceStatuses({ participantId });
     logger.info({
-      action: 'ros-status-get',
+      action: actionName,
       performed_by: user,
       ids: resp.map((r) => r.id),
     });
     return res.status(200).json(resp);
   })
 );
 
+// Update return of service fields
+router.patch(
+  '/participant/:participantId',
+  applyMiddleware(keycloak.allowRolesMiddleware('ministry_of_health')),
+  asyncMiddleware(async (req, res) => {
+    const actionName = 'ros-status-update';
+    const { participantId } = req.params;
+    const { email, user_id: userId, sub: localUserId } = req.user;
+    const user = userId || localUserId;
+    if (!(email && user)) {
+      return res.status(401).send('Unauthorized user');
+    }
+    const participant = await validateParticipant(participantId, actionName);
+    if (!participant) {
+      return res.status(404).send('Participant not found');
+    }
+    const { newValue, fieldType } = req.body;
+
+    try {
+      let response;
+      switch (fieldType) {
+        case rosFieldUpdate.siteId:
+          response = await updateReturnOfServiceSite({
+            participantId: participant.id,
+            newSiteId: newValue,
+          });
+          break;
+
+        case rosFieldUpdate.date:
+          response = await updateReturnOfServiceDate({
+            participantId: participant.id,
+            newDate: newValue,
+          });
+          break;
+
+        case rosFieldUpdate.startDate:
+          response = await updateReturnOfServiceStartDate({
+            participantId: participant.id,
+            newStartDate: newValue,
+          });
+          break;
```

Comment:
I'm not convinced we need three methods for these edits, could we combine them using a pattern like

```
validateSite(req.body.site);
validateStartDate(req.body.startDate);
const edited = {
  ...old,
  date: req.body.date || old.date,
  startDate: req.body.startDate || old.startDate,
  site: req.body.site || old.site,
}
db.save(edited);
```

Reply:
I don't think we can validate all fields by default. Only 1 field is being passed, so the validation for others will fail

Reply:
If you mean the validation on the functions themselves, like
```
if (!newSiteId) { throw new Error; }
```

I do think the right way to do it is a single `updateRosStatus` change. I'm not sure if Massive works by saving entities, but my thoughts are something like:
```
updateReturnOfService = async ({ participantId, newValues }) => {
_const existingRos = find participant ROS by participant id_
_if no participant, throw error_
_if existingRos has no delta with newValues, throw errror_
_any other checks, such as checking that for `assigned-new-site` if startDate has changed_
_apply new values into existingRos_
_save ROS_
}
```

Open to discussion here, but by and large, I wouldn't prefer a function for each property change

Reply:
I just realized that yesterday after positing a comment. Moving away from having multiple functions is definitely a good idea. Let me try to update that now